core/solver/multigrid.cpp

```diff
@@ -638,7 +652,31 @@ void Multigrid::generate()
             // default coarse grid solver, direct LU
             // TODO: maybe remove fixed index type
             auto gen_default_solver = [&]() -> std::unique_ptr<LinOp> {
-                // TODO: unify when dpcpp supports direct solver
+        // TODO: unify when dpcpp supports direct solver
+#if GINKGO_BUILD_MPI
+                if (gko::detail::is_distributed(matrix.get())) {
+                    using absolute_value_type = remove_complex<value_type>;
+                    return solver::Gmres<value_type>::build()
```

Comment:
this is quite similar/identical to the case for the dpcpp executor, maybe it could be extracted.